libvast/vast/concept/parseable/core/parser.hpp

```diff
@@ -57,35 +57,36 @@ struct parser_base {
     return derived().parse(f, l, a) && f == l;
   }
 
-  // FIXME: don't ignore ADL.
-  template <class Range, class Attribute = unused_type>
-  auto operator()(Range&& r, Attribute& a = unused) const
-  -> decltype(std::begin(r), std::end(r), bool()) {
-    auto f = std::begin(r);
-    auto l = std::end(r);
-    return derived().parse(f, l, a) && f == l;
-  }
-
-  // FIXME: don't ignore ADL.
-  template <class Range, class A0, class A1, class... As>
-  auto operator()(Range&& r, A0& a0, A1& a1, As&... as) const
-  -> decltype(std::begin(r), std::end(r), bool()) {
-    auto t = std::tie(a0, a1, as...);
-    return operator()(r, t);
-  }
-
-  template <class Iterator, class Attribute>
-  auto operator()(Iterator& f, const Iterator& l, Attribute& a) const
-  -> decltype(*f, ++f, f == l, bool()) {
-    return derived().parse(f, l, a);
+  template <class... TAttributes>
+  auto operator()(concepts::range auto&& r, TAttributes&... attributes) const {
```

Comment:
Can we keep the `-> bool` here? I feel like every bit helps with these highly abstract functions.

Reply:
Sure, just added it back.